xbmc/FileItem.cpp

```diff
@@ -3984,6 +3986,31 @@ VideoDbContentType CFileItem::GetVideoContentType() const
   return type;
 }
 
+bool CFileItem::IsPlayable() const
+{
+  const CFileItem item(GetItemToPlay());
+
+  // General
+  if (item.IsParentFolder())
+    return false;
+
+  // Plugins
+  if (item.IsPlugin() && item.GetProperty("isplayable").asBoolean())
```

Comment:
I am not a fan of the utils include in this file (which is already pretty massive). `IsPlayable` is a property which is already commonly used from (for example) plugins to flag the item is playable. I think you can avoid all this logic and includes if at some point in the chain (when MusicUtils or VideoUtils is used) you set the isplayable property on the item instead.

Reply:
If somehow this is quite hard to do (afaics this is for favourites mostly) I'd say another plausible alternative is to move those checks somewhere else (similar to what we do with URIUtils). Really my point is to avoid making CFileItem even bigger than it's already is.

Reply:
>  think you can avoid all this logic and includes if at some point in the chain (when MusicUtils or VideoUtils is used) you set the isplayable property on the item instead.

If setting the property has no bad side effects (some assumption that only plugins support it for example - the if plugin checks before getting the property value makes me nervous) this would be great idea. Thanks. I will think about it, but feel this changes could be really high risk.

Reply:
> afaics this is for favourites mostly

No, it is currently for favs and items provided by `CDirectoryListProvider` (Home screen directory listings as most prominent example).

Reply:
> I'd say another plausible alternative is to move those checks somewhere else (similar to what we do with URIUtils).

You mean to add another header which defines some free functions, and the implementation cpp file for this functions is okay to include MusicUtils.h and VideoUtils.h ? What would be the benefit over the current solution? MusicUtils.h and VideoUtils.h do not have heavy dependencies on other headers. 

For MusicUtils.h it is just "media/MediaType.h" besides of std includes and "media/MediaType.h" itself only includes std headers.
For VideoUtils.h its only `memory`.

Do you really have a point here?

Reply:
@enen92 I took the "URIUtils" approach - everything was too risky and/or not implementable with reasonable effort. Acceptable now?